internal/cmd/byok/command_create.go

```diff
@@ -0,0 +1,219 @@
+	}
+
+	cmd.Flags().String("key-vault", "", "The ID of the Azure Key Vault where the key is stored.")
+	cmd.Flags().String("tenant", "", "The ID of the Azure Active Directory tenant that the key vault belongs to.")
+	pcmd.AddOutputFlag(cmd)
+
+	cmd.MarkFlagsRequiredTogether("key-vault", "tenant")
+
+	return cmd
+}
+
+func (c *command) createAwsKeyRequest(cmd *cobra.Command, keyString string) (*byokv1.ByokV1Key, error) {
+	keyReq := byokv1.ByokV1Key{
+		Key: &byokv1.ByokV1KeyKeyOneOf{
+			ByokV1AwsKey: &byokv1.ByokV1AwsKey{
+				KeyArn: keyString,
+				Kind:   "AwsKey",
+			},
+		},
+	}
+
+	return &keyReq, nil
+}
+
+func (c *command) createAzureKeyRequest(cmd *cobra.Command, keyString string) (*byokv1.ByokV1Key, error) {
+	keyVaultID, err := cmd.Flags().GetString("key-vault")
+	if err != nil {
+		return nil, err
+	}
+	tenantID, err := cmd.Flags().GetString("tenant")
+	if err != nil {
+		return nil, err
+	}
+
+	keyReq := byokv1.ByokV1Key{
+		Key: &byokv1.ByokV1KeyKeyOneOf{
+			ByokV1AzureKey: &byokv1.ByokV1AzureKey{
+				KeyId:      keyString,
+				KeyVaultId: keyVaultID,
+				TenantId:   tenantID,
+				Kind:       "AzureKey",
+			},
+		},
+	}
+
+	return &keyReq, err
+}
+
+func (c *command) create(cmd *cobra.Command, args []string) error {
+	keyString := args[0]
+	var err error
+	var keyReq *byokv1.ByokV1Key
+
+	if cmd.Flags().Changed("key-vault") && cmd.Flags().Changed("tenant") {
+		keyString = removeKeyVersionFromAzureKeyId(keyString)
+
+		keyReq, err = c.createAzureKeyRequest(cmd, keyString)
+		if err != nil {
+			return err
+		}
+	} else if isAWSKey(keyString) {
+		keyReq, err = c.createAwsKeyRequest(cmd, keyString)
+		if err != nil {
+			return err
+		}
+	} else {
+		return errors.New(fmt.Sprintf("invalid key format: %s", keyString))
+	}
+
+	key, httpResp, err := c.V2Client.CreateByokKey(*keyReq)
+	if err != nil {
+		return errors.CatchCCloudV2Error(err, httpResp)
+	}
+
+	return c.outputByokKeyDescription(cmd, &key)
+}
+
+func isAWSKey(key string) bool {
+	keyArn, err := arn.Parse(key)
+	if err != nil {
+		return false
+	}
+
+	return keyArn.Service == "kms" && strings.HasPrefix(keyArn.Resource, "key/")
+}
+
+func getPolicyCommand(key *byokv1.ByokV1Key) (string, error) {
+	switch {
+	case key.Key.ByokV1AwsKey != nil:
+		return renderAWSEncryptionPolicy(*key.Key.ByokV1AwsKey.Roles)
+	case key.Key.ByokV1AzureKey != nil:
+		return renderAzureEncryptionPolicy(key)
+	default:
+		return "", nil
+	}
+}
+
+func renderAWSEncryptionPolicy(roles []string) (string, error) {
+	buf := new(bytes.Buffer)
+	if err := encryptionKeyPolicyAWS.Execute(buf, roles); err != nil {
+		return "", errors.New(errors.FailedToRenderKeyPolicyErrorMsg)
+	}
+	return buf.String(), nil
+}
+
+func renderAzureEncryptionPolicy(key *byokv1.ByokV1Key) (string, error) {
+	objectId := fmt.Sprintf(`$(az ad sp show --id "%s" --query id --out tsv 2>/dev/null || az ad sp create --id "%s" --query id --out tsv)`, *key.Key.ByokV1AzureKey.ApplicationId, *key.Key.ByokV1AzureKey.ApplicationId)
```

Comment:
FYI, this only works for macOS/Linux and some of our customers use Windows

Reply:
```suggestion
	objectId := fmt.Sprintf(`$(az ad sp show --id "%s" --query id --out tsv 2>/dev/null || az ad sp create --id "%s" --query id --out tsv)`, key.Key.ByokV1AzureKey.GetApplicationId(), key.Key.ByokV1AzureKey.GetApplicationId())
```

Reply:
Tracking with JIra Ticket STORAGE-307.